### PR TITLE
TST: mark failing BigQuery tests as xfail

### PR DESCRIPTION
**Pull request by tswast**
It appears that most of these issues are due to changes in how a
DataFrame is created in the `google-cloud-bigquery` library. Now that
Arrow is used as an intermediate format, arrays and structs come back as
numpy arrays or Arrow objects. Previously they came back as Python
objects (lists and dictionaries).

Towards #2353

### CODE REVIEW

File: ibis/bigquery/tests/test_client.py [hunk 1]
```diff
@@ -359,6 +359,7 @@ def test_scalar_param_date(alltypes, df, date_value):
     tm.assert_frame_equal(result, expected)
 
 
+@pytest.mark.xfail  # Issue 2374
```
Comment (jreback) [inline ibis/bigquery/tests/test_client.py:362]: can you add reason= here with the issue number, also add strict=True

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by datapythonista — APPROVED** [review]
Thanks @tswast lgtm.

Should this fix the bigquery build? Or more work is needed?

**Comment by tswast** [comment]
It should "fix" the build for all the issues you identified in #2353. 

I haven't investigated all the new failures around floating point values yet https://github.com/ibis-project/ibis/issues/2353#issuecomment-692067685

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @tswast obviously happy to have patches for these.

**Pull request merged**